src/React.re

```diff
@@ -419,6 +419,76 @@ external useImperativeHandle7:
   unit =
   "useImperativeHandle";
 
+module Uncurried = {
+  [@bs.module "react"]
+  external useState:
+    ([@bs.uncurry] (unit => 'state)) =>
+    ('state, (. ('state => 'state)) => unit) =
+    "useState";
+
+  [@bs.module "react"]
+  external useReducer:
+    ((. 'state, 'action) => 'state, 'state) => ('state, (. 'action) => unit) =
+    "useReducer";
+
+  [@bs.module "react"]
+  external useReducerWithMapState:
+    (
+      [@bs.uncurry] (('state, 'action) => 'state),
+      'initialState,
+      'initialState => 'state
```

Comment:
Unrelated to your PR but I think this is unsafe. Should probably add bs.uncurry here as well.

Reply:
Do you think we should go ahead and add `[@bs.uncurry]` to that argument on the regular `useReducerWithMapState` too?

Reply:
Yes. As a separate PR that would be great.